action-api/src/main/java/com/powsybl/action/json/ActionListDeserializer.java

```diff
@@ -29,9 +35,11 @@ public ActionListDeserializer() {
         super(ActionList.class);
     }
 
-    private static class ParsingContext {
+    public static class ParsingContext {
         String version;
         List<Action> actions;
+        Map<String, NetworkElementIdentifier> elementIdentifierMap = new HashMap<>();
+        Map<String, Action> actionBuilderMap = new HashMap<>();
```

Comment:
I would delete the build() call in the deserializers in order to get the builders here (and in the actions variable) and only build the actions if you return the list of actions. That would allow you to keep the elementsId required

Reply:
ok